iam/x509-identity-mgmt/js/config/default.conf

```diff
@@ -312,3 +312,98 @@ logger.file.max:string=7d
+# =======================================
+
+# ---------------------------
+# X509IDMGMT_KAFKA_TOPIC_ACKS
+# ---------------------------
+kafka.topic.acks:integer=-1
+
+# ----------------------------------------
+# X509IDMGMT_KAFKA_TOPIC_AUTO_OFFSET_RESET
+# ----------------------------------------
+kafka.topic.auto.offset.reset:string=latest
+
+# -----------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_CLIENT_ID
+# -----------------------------------
+kafka.consumer.client.id:string=${X509IDMGMT_APP_HOSTNAME:-x509-identity-mgmt}
+
+# ----------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_GROUP_ID
+# ----------------------------------
+kafka.consumer.group.id:string=x509-identity-mgmt-group
+
+# ----------------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_MAX_IN_FLIGHT_REQ_PER_CONN
+# ----------------------------------------------------
+kafka.consumer.max.in.flight.req.per.conn:integer=1000000
+
+# ----------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_METADATA_BROKER_LIST
+# ----------------------------------------------
+kafka.consumer.metadata.broker.list:string=127.0.0.1:9092
+
+# -------------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_SOCKET_KEEPALIVE_ENABLE
+# -------------------------------------------------
+kafka.consumer.socket.keepalive.enable:boolean=false
+
+# --------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_COMMIT_INTERVAL_MS
+# --------------------------------------------
+kafka.consumer.commit.interval.ms:integer=5000
+
+# -------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_INPROCESS_MAX_MSG
+# -------------------------------------------
+kafka.consumer.inprocess.max.msg:integer=1
+
+# ----------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_QUEUED_MAX_MSG_BYTES
+# ----------------------------------------------
+kafka.consumer.queued.max.msg.bytes:integer=10485760
+
+# -----------------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_SUBSCRIPTION_BACKOFF_MIN_MS
+# -----------------------------------------------------
+kafka.consumer.subscription.backoff.min.ms:integer=1000
+
+# -----------------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_SUBSCRIPTION_BACKOFF_MAX_MS
+# -----------------------------------------------------
+kafka.consumer.subscription.backoff.max.ms:integer=60000
+
+# -------------------------------------------------------
+# X509IDMGMT_KAFKA_CONSUMER_SUBSCRIPTION_BACKOFF_DELTA_MS
+# -------------------------------------------------------
+kafka.consumer.subscription.backoff.delta.ms:integer=1000
+
+# ******************************************************************************
+
+# =====================
+# DeviceManager Service
+# =====================
+
+# -------------------------------
+# X509IDMGMT_DEVICEMGR_DEVICE_URL
+# -------------------------------
+devicemgr.device.url:string=http://127.0.0.1:5000/device
```

Comment:
Do you really need the device-manager? Is it just to ensure that the device actually exists on that tenant and is accessible by the device manager? Is that really important? I imagine it is important because it is a security service, but I am confirming ...

Reply:
The _x509-identity-mgmt_ has a small database that lists the devices and their respective tenants. This database is populated primarily according to the events that _Device Manager_ publishes on _Kafka_. If something unexpected occurs and _x509-identity-mgmt_ is unable to populate the database from _Kafka_, it will attempt to communicate directly with _Device Manager_ as _plan B_ (and then populate the database) before refusing to associate the device with the certificate.